compiler/src/main/kotlin/dev/zacsweers/metro/compiler/fir/generators/ContributionsFirGenerator.kt

```diff
@@ -181,7 +227,21 @@ internal class ContributionsFirGenerator(session: FirSession) :
           }
         }
       }
-      .apply { markAsDeprecatedHidden(session) }
+      .apply {
+        markAsDeprecatedHidden(session)
+        val metroContributionAnnotation =
+          buildMetroContributionAnnotation().apply {
+            replaceArgumentMapping(
+              buildAnnotationArgumentMapping {
+                val originalScopeArg =
+                  contributingClassToScopedContributions.getValueIfComputed(owner)?.get(name)
+                    ?: error("Could not find a contribution scope for ${owner.classId}.$name")
+                this.mapping.put("scope".asName(), originalScopeArg)
+              }
+            )
+          }
+        replaceAnnotations(annotations + listOf(metroContributionAnnotation))
```

Comment:
We can also add this in IR if that's easier

Reply:
Ah wait I'm guessing it needs to be done here since here is where we have the tracked context 👍